src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -196,6 +199,59 @@ public SqlConnection(string connectionString, SqlCredential credential) : this()
             //      checking pool groups which is not necessary. All necessary operation is already done by calling "ConnectionString = connectionString"
         }
 
+        /// <include file='../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/ctorConnectionStringCertificate/*' />
+        public SqlConnection(string connectionString, X509Certificate clientCertificate) : this()
+        {
+            //TODO: Make constructor for SqlConnection with ClientCertificate
+            ConnectionString = connectionString;
+            if (clientCertificate != null)
+            {
+                // The following checks are necessary as setting Credential property will call CheckAndThrowOnInvalidCombinationOfConnectionStringAndSqlCredential
+                //  CheckAndThrowOnInvalidCombinationOfConnectionStringAndSqlCredential it will throw InvalidOperationException rather than ArgumentException
+                //  Need to call setter on Credential property rather than setting _credential directly as pool groups need to be checked
+                SqlConnectionString connectionOptions = (SqlConnectionString)ConnectionOptions;
+                //TODO: Update check functions
+               /* if (UsesClearUserIdOrPassword(connectionOptions))
```

Comment:
Why is this commented?

Reply:
It needs popper throw messages for use case with Clientcertificate and not error messages for Password case